src/test/java/FreedomPointsCounterTests.java

```diff
@@ -0,0 +1,99 @@
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.Scanner;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class FreedomPointsCounterTests {
+
+    private static @NotNull Board parseBoardFromString(@NotNull String printedBoard, int numberOfRows, int numberOfColumns) {
+        Scanner scanner = new Scanner(printedBoard);
+        Board board = new Board(numberOfRows, numberOfColumns);
+        while (scanner.hasNextLine()) {
+            if (scanner.hasNextInt()) {
+                int currentRow = scanner.nextInt();
+                for (int currentColumn = 1; currentColumn <= numberOfColumns; currentColumn++) {
+                    String placeholder = scanner.next("[WB-]");
+                    if (placeholder.equals("W")) {
+                        board.putStone(new Position(currentRow, currentColumn), Stone.Color.WHITE);
+                    } else if (placeholder.equals("B")) {
+                        board.putStone(new Position(currentRow, currentColumn), Stone.Color.BLACK);
+                    }
+                }
+                scanner.nextLine();
+            } else {
+                scanner.close();
+                break;
+            }
+        }
+        return board;
+    }
+
+    @Test
+    void testBoardParsing() {
```

Comment:
Should add more at least one more test for a different `Board` size

Reply:
Yes of course